qtodotxt/ui/controllers/menu_controller.py

```diff
@@ -23,13 +27,50 @@ def _initFileMenu(self):
         fileMenu = self._menu.addMenu('&File')
         fileMenu.addAction(self._createNewAction())
         fileMenu.addAction(self._createOpenAction())
+
+        lastOpened = fileMenu.addMenu("Open &Recent")
+        for ind in range(self.maxRecentFiles):
+            self.recentFileArray.append(QtWidgets.QAction(self, visible=False, triggered=self.openRecentFile))
+            lastOpened.addAction(self.recentFileArray[ind])
+
         fileMenu.addAction(self._createSaveAction())
         fileMenu.addAction(self._createRevertAction())
         fileMenu.addSeparator()
         fileMenu.addAction(self._createPreferenceAction())
         fileMenu.addSeparator()
         fileMenu.addAction(self._createExitAction())
 
+    def updateRecentFileActions(self):
+        recentFileNames = self.getRecentFileNames()
+        ind = 1
+        for i in range(len(recentFileNames)):
+            try:
+                exist = os.path.isfile(recentFileNames[i])
+            except:
+                exist = False
+            if (recentFileNames[i] is None) or (not exist):
+                self.recentFileArray[i].setVisible(False)
+                continue
+            text = "&%d %s" % (ind, recentFileNames[i])
+            self.recentFileArray[i].setText(text)
+            self.recentFileArray[i].setData(recentFileNames[i])
+            self.recentFileArray[i].setVisible(True)
+            self.recentFileArray[i].setShortcuts(["Ctrl+" + str(ind)])
+            ind += 1
+
+    def getRecentFileNames(self):
+        recentFileNames = []
+        recentFileNames.append(QtCore.QSettings().value("lastOpened", []))
+        recentFileNamesStr = []
+        for name in recentFileNames[0]:
+            recentFileNamesStr.append(name)
```

Comment:
recentFileNamesStr = [name for name in recentFileNames[0]]

but why not return recentFileNames[0] directly?